actors/migration/nv9/util.go

```diff
@@ -0,0 +1,118 @@
+package nv9
+
+import (
+	"bytes"
+	"context"
+
+	amt2 "github.com/filecoin-project/go-amt-ipld/v2"
+	amt3 "github.com/filecoin-project/go-amt-ipld/v3"
+	hamt2 "github.com/filecoin-project/go-hamt-ipld/v2"
+	hamt3 "github.com/filecoin-project/go-hamt-ipld/v3"
+	cid "github.com/ipfs/go-cid"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	adt2 "github.com/filecoin-project/specs-actors/v2/actors/util/adt"
+)
+
+// Migrates a HAMT from v2 v3 without re-encoding keys or values. The new hamt
+// configuration is specified by newOpts.
+func migrateHAMTRaw(ctx context.Context, store cbor.IpldStore, root cid.Cid, newOpts []hamt3.Option) (cid.Cid, error) {
```

Comment:
Varargs will probably help callers here.

Reply:
I changed it to accept the bitwidth explicitly as an int, and construct the args inside here.